File: hubarena/backend/app/models/court.py
from datetime import datetime
from app.database.db import db


class Court(db.Model):
    __tablename__ = "courts"

    id = db.Column(db.Integer, primary_key=True)

    arena_id = db.Column(
        db.Integer,
        db.ForeignKey("arenas.id"),
        nullable=False
    )

    sport = db.Column(db.String(80), nullable=False)
    price_hour = db.Column(db.Float, nullable=False)
    capacity = db.Column(db.Integer, nullable=False)
    available = db.Column(db.Boolean, default=True)

    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    arena = db.relationship("Arena", back_populates="courts")

    reservations = db.relationship(
        "Reservation",
        back_populates="court",
        cascade="all, delete-orphan"
    )

    def to_dict(self):
        return {
            "id": self.id,
            "arenaId": self.arena_id,
            "sport": self.sport,
            "priceHour": self.price_hour,
            "capacity": self.capacity,
            "available": self.available,
            "createdAt": self.created_at.isoformat() if self.created_at else None
        }

File: hubarena/mobile/cliente/lib/data/models/court.dart
class Court {
  final int id;
  final int arenaId;
  final String sport;
  final double priceHour;
  final int capacity;
  final bool available;

  Court({
    required this.id,
    required this.arenaId,
    required this.sport,
    required this.priceHour,
    required this.capacity,
    required this.available,
  });

  factory Court.fromJson(Map<String, dynamic> json) {
    return Court(
      id: json['id'] as int,
      arenaId: json['arenaId'] as int,
      sport: json['sport'] as String,
      priceHour: (json['priceHour'] as num).toDouble(),
      capacity: json['capacity'] as int,
      available: json['available'] as bool,
    );
  }
}
